tests/functional/plugin_utils.py

```diff
@@ -201,131 +239,153 @@ def compose_name(adcm_object: AnyADCMObject) -> str:
+    and saving it to dict of following structure:
+    {
+        Cluster: {cluster_id: value},
+        Service: {service_id: value},
+        Component: {component_id: value},
+        Provider: {provider_id: value},
+        Host: {host_id: value},
+    }
     """
-    uploaded_bundle = adcm_client.upload_from_fs(plugin_utils.get_data_dir(caller_file, bundle_dir))
-    first_provider, second_provider, *_ = [uploaded_bundle.provider_create(name=name) for name in DEFAULT_OBJECT_NAMES]
-    providers = (first_provider, second_provider)
-    for provider in providers:
-        for suffix in DEFAULT_OBJECT_NAMES:
-            provider.host_create(fqdn=f'{provider.name}-{suffix}')
-    return providers
+    frozen_objects = {cls: {} for cls in ADCMObjects}
+    for cluster in adcm_client.cluster_list():
+        if cluster.id not in to_ignore[Cluster]:
+            frozen_objects[Cluster][cluster.id] = get_attribute_func(cluster)
+        for service in cluster.service_list():
+            if service.id not in to_ignore[Service]:
+                frozen_objects[Service][service.id] = get_attribute_func(service)
+            for component in service.component_list():
+                if component.id not in to_ignore[Component]:
+                    frozen_objects[Component][component.id] = get_attribute_func(component)
+    for provider in adcm_client.provider_list():
+        if provider.id not in to_ignore[Provider]:
+            frozen_objects[Provider][provider.id] = get_attribute_func(provider)
+        for host in provider.host_list():
+            if host.id not in to_ignore[Host]:
+                frozen_objects[Host][host.id] = get_attribute_func(host)
+    return frozen_objects
+
+
+def run_successful_task(action: Action, action_owner_name: str):
+    """Run action and expect it succeeds"""
+    task = action.run()
+    try:
+        wait_for_task_and_assert_result(task, status="success")
+    except AssertionError as error:
+        raise AssertionError(
+            f'Action {action.name} should have succeeded when ran on {action_owner_name}:\n' f'{error}'
+        ) from error
+
+
+def __check_components(
+    adcm_client: ADCMClient,
+    comparator: Callable,
+    components: Dict[int, ADCMObjectField],
+    service_ids: Collection[int],
+):
+    """Check components config is intact since .component is not implemented"""
+    component_ids = components.keys()
+    for service in (adcm_client.service(id=sid) for sid in service_ids):
+        for component in service.component_list():
+            if (component_id := component.id) in component_ids:
+                comparator(component, components[component_id])
+
+
+def __build_ignore_map(objects_to_ignore: Collection[AnyADCMObject]):
+    """Returns map with ids of objects to ignore grouped by class"""
+    # may be empty
+    ignore_map = {cls: set() for cls in ADCMObjects}
+    for obj in objects_to_ignore:
+        ignore_map[obj.__class__].add(obj.id)
+    return ignore_map
```

Comment:
Double underscore looks redundant. I think one enough

Reply:
I agree that ignore map can be single underscored, as it can be more common solution, so "protected" is okay. But checking components is something that can be used in one place only, because it's sort of a crutch since client can't search compinents from "root" by id